### ISSUE TITLE
Segmented DASH stream playback stopping

### ISSUE THREAD

**Issue by SteUK** [google/ExoPlayer#1876]
Hi,

Tested in master and dev v2 branch.

I have a 1 minute DASH stream (though this happens on a 5 minute clip as well) that gets to about 30 seconds and stops playing.

The stream is fragmented MP4, segmented into 4 second chunks.  The source frame rate is 29.97 and Keyframes are every 60 frames.  There is an audio AdaptatonSet with single representation.

Not sure if the 30 seconds part is a red herring, maybe the problem, if indeed it is a bug, is when the buffer fills up with the last remaining chunk still to be loaded in.  Following this hunch I have delved into debugging the depths of the code base, trying to find out why this last chunk might not be loading.  Thus far I have tracked it into maybeContinueLoading(), where shouldContinueLoading() is called on the loadControl object.  This seems to not return true when the last buffer chunk needs to be loaded.

I will keep digging to see if I can find anything else...

Test stream to show the problem is here:  http://clients.renderheads.com/temp/BuckTest/BuckTest.mpd

Tested in ExoPlayer v1.X and it plays back to the end with no problem.

Anyone else seeing a similar issue?

**Comment by ojw28** [comment]
The manifest you link to is quite badly malformed. The main problem is the video AdaptationSet. It's claiming that each video segment is `(49152 / 12288) = 4` seconds long. There are 24 of these segments, which gives a total duration of 96 seconds. This is 36 seconds longer than the top level presentation duration of 1 minute, which shouldn't happen. Each segment actually only contains about 2.5s of video. The result is that ExoPlayer thinks it has more video buffered than it actually does, the video buffer underruns, and playback stalls. Seeking will also not function efficiently, if at all. If the manifest works in V1, this is more by chance (and it definitely wont be playing in an efficient way).

Note that the duration of audio `(192000 / 48000) * 16 = 64`, also exceeds the top level presentation duration, which again should not happen.

**Comment by SteUK** [comment]
Apologies!

Firstly, I had failed to notice the big buck sample I converted was 24fps, rather than the 29.97fps of the content I am creating (cannot post that up due to client confidentiality :I).  I ran the same test converting it up to 30fps before running it though mp4box and I am getting durations of 61440ms (!) and 17 segments.

Sooooo, just tried fragmenting and splitting the same source mp4 video using the bento4 set of tools...and I have 15 segments!  Not tried it yet, but signs are good.  I guess this is actually issues(s) with mp4box.  Sorry to waste your time, but maybe this will serve as a warning in the short term for anyone else that hits this issue.
